packages/core/src/routes/account/mfa-verifications.ts

```diff
@@ -100,60 +108,110 @@ export default function mfaVerificationsRoutes<T extends UserRouter>(
       const { mfa } = await findDefaultSignInExperience();
       assertThat(mfa.factors.includes(ctx.guard.body.type), 'session.mfa.mfa_factor_not_enabled');
 
-      if (ctx.guard.body.type === MfaFactor.TOTP) {
-        // A user can only bind one TOTP factor
-        assertThat(
-          user.mfaVerifications.every(({ type }) => type !== MfaFactor.TOTP),
-          new RequestError({
-            code: 'user.totp_already_in_use',
-            status: 422,
-          })
-        );
-
-        // Check secret
-        assertThat(validateTotpSecret(ctx.guard.body.secret), 'user.totp_secret_invalid');
-
-        const updatedUser = await updateUserById(userId, {
-          mfaVerifications: [
-            ...user.mfaVerifications,
-            {
-              id: generateStandardId(),
-              createdAt: new Date().toISOString(),
-              type: MfaFactor.TOTP,
-              key: ctx.guard.body.secret,
-            },
-          ],
-        });
-
-        ctx.appendDataHookContext('User.Data.Updated', { user: updatedUser });
-
-        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-      } else if (ctx.guard.body.type === MfaFactor.WebAuthn) {
-        const { newIdentifierVerificationRecordId, name } = ctx.guard.body;
-        // Check new identifier
-        const newVerificationRecord = await buildVerificationRecordByIdAndType({
-          type: VerificationType.WebAuthn,
-          id: newIdentifierVerificationRecordId,
-          queries,
-          libraries,
-        });
-        assertThat(newVerificationRecord.isVerified, 'verification_record.not_found');
-
-        const bindMfa = newVerificationRecord.toBindMfa();
-
-        const updatedUser = await updateUserById(userId, {
-          mfaVerifications: [
-            ...user.mfaVerifications,
-            {
-              ...bindMfa,
-              id: generateStandardId(),
-              createdAt: new Date().toISOString(),
-              name,
-            },
-          ],
-        });
-
-        ctx.appendDataHookContext('User.Data.Updated', { user: updatedUser });
+      switch (ctx.guard.body.type) {
```

Comment:
Should we split this into multiple APIs?  `${accountApiPrefix}/mfa-verifications/:type`

Reply:
This is aligned with Management API `POST /users/:userId/mfa-verifications`